bindings/rust/s2n-tls-sys/build.rs

```diff
@@ -160,14 +184,23 @@ fn build_cmake() {
 
     let dst = config.build();
 
-    // tell rust we're linking with libcrypto
-    let root = PathBuf::from(env("DEP_OPENSSL_ROOT"));
-    if root.join("libcrypto.so").exists() {
-        println!("cargo:rustc-link-lib=crypto");
+    let lib = search(dst.join("lib64"))
+        .or_else(|| search(dst.join("lib")))
+        .or_else(|| search(dst.join("build").join("lib")))
+        .expect("could not build libs2n");
+
+    // link the built artifact
+    if lib.join("libs2n.a").exists() {
```

Comment:
Why switch from checking for the .so to the .a? (see old line 186)

Reply:
I realized `.so` is only Linux. On macOS it's going to by `.dylib`. Whereas if we use `.a` we don't have to care :)